irctest/server_tests/utf8.py

```diff
@@ -11,30 +10,38 @@
 
 class Utf8TestCase(cases.BaseServerTestCase):
     @cases.mark_specifications("Ergo")
-    def testUtf8Validation(self):
+    def testNonUnicodeFiltering(self):
         self.connectClient(
             "bar",
             capabilities=["batch", "echo-message", "labeled-response"],
         )
         self.joinChannel(1, "#qux")
-        self.sendLine(1, "PRIVMSG #qux hi")
-        ms = self.getMessages(1)
-        self.assertMessageMatch(
-            [m for m in ms if m.command == "PRIVMSG"][0], params=["#qux", "hi"]
-        )
-
-        self.sendLine(1, b"PRIVMSG #qux hi\xaa")
+        self.sendLine(1, b"@label=xyz PRIVMSG #qux hi\xaa")
         self.assertMessageMatch(
             self.getMessage(1),
             command="FAIL",
             params=["PRIVMSG", "INVALID_UTF8", ANYSTR],
-            tags={},
+            tags={"label": "xyz"},
         )
 
-        self.sendLine(1, b"@label=xyz PRIVMSG #qux hi\xaa")
+    @cases.mark_isupport("UTF8ONLY")
+    @cases.mark_capabilities("echo-message")
```

Comment:
Could you change this test to not depend on `echo-message`? Instead, you can connect another user and make sure they didn't receive any message.